### PR TITLE
Add TEXTJOIN function

### PR DESCRIPTION
**Pull request by marcin-kordas-hoc**
## Summary

- Implements `TEXTJOIN(delimiter, ignore_empty, text1, [text2, …])` in `TextPlugin`
- Supports scalar and array/range delimiters (with cycling when delimiter is a range)
- Skips empty strings when `ignore_empty` is `TRUE`; includes them when `FALSE`
- Returns `#VALUE!` when the result exceeds 32,767 characters (Excel cell limit)
- Adds `ErrorMessage.TextJoinResultTooLong` for the length-exceeded error
- Adds translations for all 17 supported languages
- Documents the function in `docs/guide/built-in-functions.md`

## Test plan

- [ ] 22 unit tests in `test/textjoin.spec.ts` covering basic usage, `ignore_empty` behaviour, array/range delimiters with cycling, type coercion, error propagation, and edge cases
- [ ] Smoke tests in `test/smoke.spec.ts` include TEXTJOIN scenarios
- [ ] All 41 tests pass (`npm run test:jest`)
- [ ] TypeScript compiles without errors (`npm run compile`)
- [ ] No lint errors (`npm run lint`)

---

> [!NOTE]
> **Medium Risk**
> Adds a new core formula function in the interpreter (`TEXTJOIN`) with range handling, delimiter cycling, and a new #VALUE! error path, which could affect calculation behavior in edge cases. Changes are additive and well-covered by new unit/smoke tests.
> 
> **Overview**
> Adds support for the `TEXTJOIN(delimiter, ignore_empty, text1, [text2, …])` formula, including scalar or range/array delimiters (cycled between joined parts), optional skipping of empty strings, and propagation of cell errors from delimiter/text inputs.
> 
> Introduces a new `ErrorMessage.TextJoinResultTooLong` and returns `#VALUE!` when the joined result exceeds 32,767 characters. Updates built-in function documentation, adds `TEXTJOIN` translations across supported languages, and adds dedicated `TEXTJOIN` unit tests plus smoke coverage.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit cfa61b5ad08c23f2ee111fd187b4710943603619. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: test/smoke.spec.ts [hunk 1]
```diff
@@ -63,6 +63,102 @@ describe('HyperFormula', () => {
     hf.destroy()
   })
 
+  it('should evaluate TEXTJOIN with ignore_empty=TRUE', () => {
```
Comment (sequba) [inline test/smoke.spec.ts]: Do not add tests to this file. We want to keep it very basic. Move all the tests to `textjoin.spec.ts`

**Review by sequba — COMMENTED** [review]
Good job

File: test/textjoin.spec.ts [hunk 2]
```diff
@@ -0,0 +1,309 @@
+import {HyperFormula} from '../src'
```
Comment (sequba) [inline test/textjoin.spec.ts:1]: @marcin-kordas-hoc Please, move tests to hyperformula-test repository

**Review by sequba — CHANGES_REQUESTED** [review]

**Comment by marcin-kordas-hoc** [comment]
Closing in favor of a new PR from upstream branch (handsontable/hyperformula:feature/TEXTJOIN_w_docs). Review feedback addressed: tests moved to private repo, docs ordering fixed, unsafe cast fixed.
